### ISSUE TITLE
Make Q API more intuitive and less error prone

### ISSUE THREAD

**Issue by mysticfall** [greencatsoft/scalajs-angular#23]
Currently, ```$http.get``` returns ```HttpPromise``` instance, which can be implicitly converted to Scala's ```Future``` as follows:

```scala
    val future: Future[User] = http.get("/users/john")

    future onComplete {
      case Success(user) => scope.id = user.id
      case Failure(t) => handleError(t)
    }
```

If you try to simplify it as below code, it'll fail at _runtime_, because it fails to infer correct type of the ```Future``` instance:

```scala
    http.get("/users/john") onComplete {
      case Success(user) => scope.id = user.id
      case Failure(t) => handleError(t)
    }
```

This is unfortunate, and we shouldn't enforce users to write in any particular style, or at least, warn them at compile time.

Further more, currently both ```Promise``` and ```HttpPromise``` provide implicit conversion method ```promise2future``` which is quite confusing and error prone.

If you use the wrong import statement, it will again fail at _runtime_ with an obscure error message. Also, it could be a problem when you need both types of conversions in the same source file.

**Comment by math85360** [comment]
I think this issue was solved with #50 ? @mysticfall

**Comment by mysticfall** [comment]
@math85360 Yeah, it should be. I almost forgot :) I'll close it as soon as I test it. Thanks!
